app/src/views/features/sessions/SessionViewer/SessionDetails.tsx

```diff
@@ -143,12 +162,71 @@ const SessionDetails: React.FC<SessionDetailsProps> = ({ isInsideIframe = false
     };
   }, [player]);
 
-  useEffect(() => {
-    player?.addEventListener("ui-update-current-time", ({ payload }) => {
-      currentTimeRef.current = startTime + payload;
+  const updateCurrentTimeHandler = useCallback(
+    ({ payload }: { payload: number }) => {
+      const currentTime = startTime + payload;
+      currentTimeRef.current = currentTime;
       setPlayerTimeOffset(payload / 1000); // millis -> secs
-    });
-  }, [player, startTime]);
+
+      if (!isSkippingInactivity.current && skipInactiveSegments.current) {
+        const skipEvent = inactiveSegments?.find(([startTime, endTime]) => {
+          return currentTime >= startTime && currentTime < endTime - 2000;
+        });
+
+        if (skipEvent) {
+          setPlayerState(PlayerState.SKIPPING);
+          isSkippingInactivity.current = true;
+
+          setTimeout(() => {
+            player.goto(skipEvent[1] - startTime - 2000);
+            setPlayerState(PlayerState.PLAYING);
+            isSkippingInactivity.current = false;
+          }, 2000);
+        }
+      }
+    },
+    [inactiveSegments, player, startTime]
+  );
+
+  const eventCastHandler = useCallback(
+    (event: RRWebEventData) => {
+      if (!skipInactiveSegments.current) return;
+      if (isSkippingInactivity.current) return;
+      // rrweb adds a delay to the events, so we need to add that delay to the timestamp to compare with the current time
+      if (event.timestamp + (event.delay ?? 0) < currentTimeRef.current) return;
+
+      const skipEvent = inactiveSegments?.find(([startTime]) => {
+        return event.timestamp === startTime;
+      });
```

Comment:
Do we know any approx number for how many times the `eventCastHandler ` will be called and the maximum inactiveSegments length can be? Since for each `event-cast` event triggered `inactiveSegments?.find()` will iterate linearly and can affect the performance.

Reply:
On paper, the max length of `inactiveSegments` can be 60 for a 10 min session.